### ISSUE TITLE
Grafana loki grafana-agent-test-0  Privileged container is not allowed

### ISSUE THREAD

**Issue by alkdese** [grafana/agent#1928]
I have deployed grafana loki simple distributed 1.7.4 and I'm getting this error

```
create Pod grafana-agent-test-0 in StatefulSet grafana-agent-test failed error: admission webhook "validation.gatekeeper.sh" denied the request: [azurepolicy-k8sazurecontainernoprivilege-zzzzzzzzz] Privileged container is not allowed: config-reloader, securityContext: {"privileged": true, "runAsUser": 0}
```

looked through the values and doco https://grafana.com/docs/loki/latest/installation/simple-scalable-helm/ none of them mentions privileged containers why it is required. 

Is there a workaround? I tend to believe if I disable 'monitoring.selfMonitoring.enable' but ideally I would like to avoid doing that.

**Comment by rfratto** [comment]
👋 This is something we've investigated a few times in the past, but don't have a clear answer on yet. We inherited `runAsUser: 0` and `privileged: true` from Promtail. 

The understanding is that you _at least_ need `runAsUser: 0` to be guaranteed to be able to read logs which are mounted from the host. We're unsure if `privileged: true` is necessary.

The container mentioned in your log line (`config-reloader`), definitely doesn't need privileged, though; just the agent pods. 

We don't maintain the Helm chart you're using, so I'm not sure if there's a way to easily disable privileged mode to check if everything still works.
